pkg/agent/run.go

```diff
@@ -130,19 +131,31 @@ func run(ctx context.Context, cfg cmds.Agent, proxy proxy.Proxy) error {
 		}
 	}
 
-	notifySocket := os.Getenv("NOTIFY_SOCKET")
-	os.Unsetenv("NOTIFY_SOCKET")
-
-	if err := setupTunnelAndRunAgent(ctx, nodeConfig, cfg, proxy); err != nil {
-		return err
+	if nodeConfig.Docker {
+		if err := executor.Docker(ctx, nodeConfig); err != nil {
+			return err
+		}
+	} else if nodeConfig.ContainerRuntimeEndpoint == "" {
+		if err := containerd.SetupContainerdConfig(nodeConfig); err != nil {
+			return err
+		}
+		if err := executor.Containerd(ctx, nodeConfig); err != nil {
+			return err
+		}
 	}
-
 	// the agent runtime is ready to host workloads when containerd is up and the airgap
```

Comment:
```suggestion
	// the container runtime is ready to host workloads when containerd is up and the airgap
```

Reply:
And also at https://github.com/k3s-io/k3s/blob/1c1746114c94498f8b77e0ba13c43ebe6e0da56a/pkg/cli/server/server.go#L113-L117